File: frontend/src/types/track.ts
export interface Track {
  id: string;
  title: string;
  artist?: string;
  album?: string;
  genre?: string;
  year?: number;
  duration: number;
  file_path: string;
  file_size: number;
  format: string;
  bitrate: number;
  waveform_data?: number[];
  created_at: string;
  updated_at: string;
}

export interface TrackCreate {
  title: string;
  artist?: string;
  album?: string;
  genre?: string;
  year?: number;
  file: File;
}

export interface TrackUpdate {
  title?: string;
  artist?: string;
  album?: string;
  genre?: string;
  year?: number;
} 

File: backend/app/schemas/track.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field


# Общие свойства
class TrackBase(BaseModel):
    title: Optional[str] = None
    artist: Optional[str] = None
    album: Optional[str] = None
    genre: Optional[str] = None
    year: Optional[int] = None


# Свойства для создания трека
class TrackCreate(TrackBase):
    title: str
    file_path: str
    duration: float


# Свойства для обновления трека
class TrackUpdate(TrackBase):
    is_active: Optional[bool] = None


# Свойства, возвращаемые клиенту
class Track(TrackBase):
    id: int
    file_path: str
    file_size: Optional[int] = None
    duration: float
    format: Optional[str] = None
    bitrate: Optional[int] = None
    waveform_data: Optional[str] = None
    
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
    user_id: int
    
    class Config:
        from_attributes = True


# Расширенная информация о треке
class TrackDetail(Track):
    pass 